components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -1446,6 +1446,12 @@ AtmosphereOutput::create_diagnostic (const std::string& diag_field_name) {
     } else {
       params.set<std::string>("Temperature Kind", "Tot");
     }
+  } else if (diag_field_name.find("_horiz_avg")!=std::string::npos) {
```

Comment:
You put this at the very last "else if" slot, so it should be ok. Nevertheless, I would like to guard against any possible future code change. In particular, if one adds handling of  "_vert_avg" , and puts an "else if" below yours, we could have a problem. Say the user asks for "some_tracer_horiz_avg_vert_avg" (yes, that's just a global absolute avg, but whatever). Your code will now just compute "some_tracer_horiz_avg", losing completely the "_vert_avg" part.

To guard against this, you can consider only the case where the field name _ends_ with "_horiz_avg". C++ doesn't have an `ends_with` method for std::string like in python (we impl-ed `starts_with` in ekat, maybe we should impl `ends_with` as well), so you'd have to roll your own. Or maybe you could do
```c++
else if (diag_field_name.find("_horiz_avg")!=std::string::npos &&
            ekat::split(diag_field_name,"_horiz_avg")[1]=="")
```
although this relies on `ekat::split` adding an empty string at the end if the string ends with the separation token.

Reply:
I realized that something like `some_tracer_horiz_avg_at_500hPa` is prob a better example for this issue.

Reply:
Actually, I take it back. `some_tracer_horiz_avg_at_500hPa` _should_ work now, since we first parse the "_at_500hPa" case, using whatever is before as the field to give to FieldAtPressureLevel. It happens to be a diag, so it creates nested diags.

Again, as of now, everything _should_ work, since the horiz_avg diag is the last one you consider. However, we should still make sure that adding a new `else if` after to handle a new diag would not mess up parsing of the diag name.

Reply:
Ah, good point. We do the same mistake with all the diags. For your diag, we can easily check that "_horiz_avg" is the last token, so that's easy. But for "blah_at_500hPa", we don't check that there's nothing after "hPa".

We can merge this, and fix ALL diags handling later, or we can fix yours here, and do a separate PR to fix other diag names that involve special tokens.

Reply:
I'm working on a cleaner solution regarding diags creation...